airflow-core/src/airflow/ui/src/pages/TaskInstances/TaskInstances.tsx

```diff
@@ -51,17 +60,55 @@ const {
   STATE: STATE_PARAM,
 }: SearchParamsKeysType = SearchParamsKeys;
 
-const taskInstanceColumns = ({
+const getColumns = ({
+  allRowsSelected,
   dagId,
+  onRowSelect,
+  onSelectAll,
   runId,
+  selectedRows,
   taskId,
   translate,
 }: {
   dagId?: string;
   runId?: string;
   taskId?: string;
   translate: TFunction;
-}): Array<ColumnDef<TaskInstanceResponse>> => [
+} & GetColumnsParams): Array<ColumnDef<TaskInstanceResponse>> => [
+  ...(Boolean(dagId)
```

Comment:
Why put a boolean on the select? I feel like people would want to perform bulk actions on the main task instance list page too?